src/linux/mod.rs

```diff
@@ -93,6 +73,62 @@ impl Sandbox for LinuxSandbox {
     }
 }
 
+/// Path permissions required for the sandbox.
+#[derive(Default)]
+pub(crate) struct PathExceptions {
+    bind_mounts: HashMap<PathBuf, MountAttrFlags>,
+    symlinks: Vec<(PathBuf, PathBuf)>,
+}
+
+impl PathExceptions {
+    /// Add or modify a path's exceptions.
+    ///
+    /// This will add a new bind mount for the canonical path with the specified
+    /// permission if it does not exist already.
+    ///
+    /// If the bind mount already exists, it will *ADD* the additional
+    /// permissions.
+    fn update(&mut self, path: PathBuf, write: bool, execute: bool) {
+        // Use canonical path for indexing.
+        //
+        // This ensures that a symlink and its target are treated like the same path for
+        // exceptions.
+        //
+        // If the home path cannot be accessed, we ignore the exception.
+        let canonical_path = match path.canonicalize() {
+            Ok(path) => path,
+            Err(_) => return,
```

Comment:
This is optional, but this path could potentially replace [the `path.exists()` check][1] in `add_exception()`. Any paths that don't exist will fail the `.canonicalize()`.

[1]: https://github.com/phylum-dev/birdcage/blob/baffbe6536fd47b226697edc1a5ee7022f1e76fc/src/linux/mod.rs#L32-L40

If you change the return of this function to `Result<()>`, you'd have to return an error here and in the other return. In both places, I think `Err(Error::InvalidPath(path.into()))` would be fine.